symphony-bdk-legacy/symphony-api-client-java/src/main/java/services/DatafeedEventsServiceV2.java

```diff
@@ -74,10 +71,19 @@ private void readEventsFromDatafeed() {
             datafeedId = datafeedIds.get(0).getId();
         }
         //Read datafeed in loop
+        logger.info("Start reading datafeed events from datafeed {}", datafeedId);
         do {
-            List<DatafeedEvent> events = datafeedClient.readDatafeed(datafeedId, datafeedClient.getAckId());
-            if (events != null && !events.isEmpty()) {
-                handleEvents(events);
+            try {
+                List<DatafeedEvent> events = datafeedClient.readDatafeed(datafeedId, datafeedClient.getAckId());
+                if (events != null && !events.isEmpty()) {
+                    handleEvents(events);
+                }
+            } catch (APIClientErrorException e) {
+                //Datafeed was stale
+                logger.debug("Something went wrong with this datafeed: {}", datafeedId);
+                logger.info(e.getMessage());
+                datafeedClient.deleteDatafeed(datafeedId);
```

Comment:
will that call always succeed? even if the data feed is faulty or not known anymore? if not don't we risk having a weird chain of logs?

also in   services.DatafeedEventsServiceV2#readDatafeed
we already had
   } catch (APIClientErrorException e) {
                    //Datafeed was stale
                    logger.info(e.getMessage());

does this make sense to have two catches with the same comment? should'nt it be only one?



Looking at that code again I asked my self should we delete the datafeed in stopDatafeedService? or at the end of readEventsFromDatafeed?

Reply:
> will that call always succeed? even if the data feed is faulty or not known anymore? if not don't we risk having a weird chain of logs?
This delete call will return 400 error if the the datafeed is not found -> APIClientError, should we add another try/catch inside catch block?

> does this make sense to have two catches with the same comment? should'nt it be only one?
I removed the one in `services.DatafeedEventsServiceV2#readDatafeed` 

>Looking at that code again I asked my self should we delete the datafeed in stopDatafeedService? or at the end of readEventsFromDatafeed?

I'm not sure about that idea, because I think the datafeed is still reusable after the service stops.

Reply:
perhaps the 400 error is something we would like to catch and just log in debug if this happens instead of waiting before retrying